source/server/CMakeLists.txt

```diff
@@ -84,7 +84,14 @@ else()
     set(SERVER_BINARY_TYPE "")
 endif()
 
-add_executable(${QFUSION_SERVER_NAME} ${SERVER_BINARY_TYPE} ${SERVER_HEADERS} ${SERVER_SOURCES} ${SERVER_PLATFORM_SOURCES})
+if (BUILD_STEAMLIB)
+    set(STEAMSHIM_FILES "../steamshim/steamshim_child.c" "../steamshim/steamshim_child.h")
+    source_group("steamshim" FILES ${STEAMSHIM_FILES})
```

Comment:
I would suggest to create a static library `steamshim_child` instead of adding the files to multiple projects like you are doing now.
Then you can use this static library to control include paths, preprocessor commands and whatnot.

Reply:
https://github.com/CoolElectronics/warfork-qfusion/commit/d7bd89e06ca3f813d8b0e894c0d41635d03b6d23

note, this is finnicky, probably not done the right way and i'm pretty sure there's bugs so someone who actually knows cmake should probably review/fix